### PR TITLE
Outdated

### PR DESCRIPTION
**Pull request by jasongin**
This is building on #73. I have enhanced the output to show both minor and patch upgrade versions when available, in a more concise format.

Example output from `nvs outdated`:
```
 node/12.18.2/x64
 node/12.16.2/x64 [~12.16.3] [^12.18.2]
 node/10.20.1/x64 [^10.21.0]
```

Also I rebased and added tests.

### CODE REVIEW

File: test/modules/listTests.js [hunk 1]
```diff
@@ -116,6 +120,7 @@ test.beforeEach(t => {
 				{ browser_download_url: downloadUrl + 'v5.6.7/node-v5.6.7-osx-x86.tar.xz' },
 				{ browser_download_url: downloadUrl + 'v5.6.7/node-v5.6.7-win-x64.7z' },
 				{ browser_download_url: downloadUrl + 'v5.6.7/node-v5.6.7-win-x86.7z' },
+
```
Comment (brodybits) [inline test/modules/listTests.js:123]:
```suggestion
```

remove extra blank line here?

**Comment by jasongin** [comment]
Merged. Thanks @rmg for the idea and initial implementation!

File: lib/list.js [hunk 2]
```diff
@@ -238,7 +270,17 @@ function listRemoteAsync(filter) {
  * @param remoteName Required name of one of the remotes configured in settings.json.
  * @returns An array of version objects.
  */
+getRemoteVersionsAsync.cache = new Map();
```
Comment (brodybits) [inline lib/list.js:273]: nitpick: I am sorry to say I think this assignment does not belong between a function and its corresponding JSDoc comments. I can only imagine what would happen if we would try running `tsc` in strict mode.

And I am now wondering why this should be a visible, undocumented member of the function in the first place. Couldn't we just define this as a `const` near the beginning, say right after `const githubReleasesRegex`?

Comment (jasongin) [reply lib/list.js:273]: Good point. I had overlooked this. Surprisingly, typescript doesn't complain about it when I run in type-checking mode against the JS code. But I'll clean it up.

Comment (jasongin) [reply lib/list.js:273]: Fixed in db5548e

**Pull request merged**